docs/architecture/adr-005-consensus-height-events.md

```diff
@@ -0,0 +1,83 @@
+# ADR 005: UpdateClient Events - ClientState Consensus Heights
+
+## Changelog
+* 25/04/2022: initial draft
+
+## Status
+
+Accepted 
+
+## Context
+
+The `ibc-go` implementation leverages the [Cosmos-SDK's EventManager](https://github.com/cosmos/cosmos-sdk/blob/main/docs/core/events.md#EventManager) to provide subscribers a method of reacting to application specific events.
+IBC relayers depend on the [`consensus_height`](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/keeper/events.go#L33) attribute emitted as part of `UpdateClient` events in order to run misbehaviour detection by cross-checking the details of the *Header* emitted at a given consensus height against that of the *Header* from the originating chain.
```

Comment:
I guess you mean that it cross-checks the details... Would you be able to specify in more details what those details are?

```suggestion
IBC relayers depend on the [`consensus_height`](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/keeper/events.go#L33) attribute emitted as part of `UpdateClient` events in order to run misbehaviour detection by cross-checking the details of the *Header* emitted at a given consensus height against those of the *Header* from the originating chain.
```

Reply:
Sure, I really paraphrased this part from https://github.com/cosmos/ibc-go/pull/1208#discussion_r851167308

But I will try to add more details here, although I think this is technically hermes specific behaviour and also required the consumer to be able to parse and unmarshal the `ClientMessage`(`Header`) from the events so perhaps it should read "Some IBC relayers depend on the consensus height...."

Reply:
Updated with additional info using [this source](https://github.com/informalsystems/ibc-rs/blob/master/modules/src/clients/ics07_tendermint/header.rs#L23)